gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/GradleCacheExplorer.java

```diff
@@ -57,17 +59,36 @@ public final Set<String> getCompletions(Set<String> repoUrls, DependencyGroup in
                 .flatMap(Optional::stream)
                 .collect(Collectors.toSet());
 
+        Set<String> resultsWithStarsIncluded = results.stream()
+                .map(result -> {
+                    int colonIndex = result.indexOf(':');
+                    if (colonIndex != -1) {
+                        return Arrays.asList(result, result.substring(0, colonIndex) + ":*");
+                    }
+                    return Collections.singletonList(result);
+                })
+                .flatMap(Collection::stream)
+                .collect(Collectors.toSet());
+
         if (parsedInput.isEmpty()) {
             return results;
         }
 
-        stopWatch.stop();
-        log.debug("Completion matching time: {} ms", stopWatch.elapsed().toMillis());
-
-        return results.stream()
+        Set<String> filteredResults = resultsWithStarsIncluded.stream()
                 .filter(result -> result.startsWith(parsedInput))
                 .map(result -> result.substring(parsedInput.length() + 1))
                 .collect(Collectors.toSet());
+
+        if (isPackageName) {
+            filteredResults = filteredResults.stream()
+                    .filter(result -> !result.contains(":"))
+                    .collect(Collectors.toSet());
+        }
```

Comment:
not the worst thing in the world, but a little strange that you're constructing and then deconstructing a set even though `isPackageName` is effectively static for the lifetime of this function.

You could add another filter to the stream above, but that seems "wasteful" in both cases i.e. if you `isPackageName` is false, you're still having that check before you get to your filter. Despite it always being `true` or `false` for the lifetime of this function.

A thing you could do is have a method: 

```java
Stream<...> packageNameFilter(Stream<...> stream, boolean isPackageName) {
  if (!isPackageName) {
    return stream;
  } else {
    return stream.filter(...);
  }
}

// ...
Set<String> filteredResults = ...;
return packageNameFilter(filteredResults, isPackageName);
```

If you use something like [StreamEx](https://github.com/amaembo/streamex) you can use their useful `chain` utility method which makes it more fluent:

```java
Set<String> filteredResults = StreamEx.of(resultsWithStarsIncluded)
  .filter(...)
  .map(...)
  .chain(packageNameFilter(...))
  .toSet();
```

Reply:
switched to using streamx